controllers/snapshot/snapshot_adapter.go

```diff
@@ -129,6 +135,72 @@ func (a *Adapter) EnsureAllIntegrationTestPipelinesExist() (results.OperationRes
 	return results.ContinueProcessing()
 }
 
+//EnsureCreationOfEnvironment makes sure that all envrionemnts that were requested via
+//IntegrationTestScenarios get created, in case that environment is already created, provides
+//a message about this fact
+func (a *Adapter) EnsureCreationOfEnvironment() (results.OperationResult, error) {
+
```

Comment:
Just a reminder to add this condition here:
```
if gitops.HaveHACBSTestsFinished(a.snapshot) {
		a.logger.Info("The Snapshot has finished testing.")
		return results.ContinueProcessing()
	}
```

Reply:
done

Reply:
Thanks!